### ISSUE TITLE
AnimatedDataSource does not update tableView when not in view hierarchy

### ISSUE THREAD

**Issue by riccardo-tesio** [RxSwiftCommunity/RxDataSources#84]
I think there is a bug in the last release of `RxTableViewSectionedAnimatedDataSource` (and maybe in `RxCollectionViewSectionedAnimatedDataSource`), when the tableView is not in view hierarchy.

Starting at line 42, the table is reloaded when not in hierarchy:
```
if tableView.window == nil {
    tableView.reloadData()
    return
}
```
but the dataSet is not updated so the table is reloaded with the old data.
Should be something like that:
```
if tableView.window == nil {
    dataSource.setSections(newSections)
    tableView.reloadData()
    return
}
```
Thanks in advance for your attention.

**Comment by kzaher** [comment]
Hi @riccardo-tesio ,

you are correct.

Feel free to create a PR.

**Comment by kzaher** [comment]
I think the same is valid for `UICollectionView`.

**Comment by pschneider** [comment]
@kzaher 
Is it somehow possible that this fix is not available in the CocoaPods 1.0.2 release? 
According to our Podfile it should be included:
```
  - RxDataSources (1.0.2):
    - RxCocoa (~> 3.0)
    - RxSwift (~> 3.0)
```
But the two lines are not included. Adding them manually fixes the mentioned problem for us.
